### PR TITLE
Make entry point requirement configurable per language.

### PR DESCRIPTION
**Pull request by thijskh**
Also allow to specify the description of the field in the UI.

Still missing: submit client changes for this.

### CODE REVIEW

File: webapp/web/js/domjudge.js [hunk 1]
```diff
@@ -403,26 +378,55 @@ function initReload(refreshtime)
...
+			case 'kt':
+				entry_point.value = entryPointDetectKt(filename);
+				break;
+			default:
+				entry_point.value = filename;
 		}
 	}
 }
 
+function entryPointDetectJava(filename)
```
Comment (eldering) [inline webapp/web/js/domjudge.js]: What about moving this into a per language code snippet in the DB or as a new type of script?

Comment (meisterT) [reply webapp/web/js/domjudge.js]: if we make this a per language code snippet that we could make the same snippet available via the API as well and use the API endpoint from both the web site and the submit client

Comment (nickygerritsen) [reply webapp/web/js/domjudge.js]: THen we do have to move it to PHP I guess, but that is possible by using some AJAX of course

Comment (thijskh) [reply webapp/web/js/domjudge.js]: Moving the detection to API calls both from commandline and web is possible but seems a big enough change vs how it works now that I'd like to move that to a separate feature request.

Comment (meisterT) [reply webapp/web/js/domjudge.js]: sounds reasonable to do this after the 6.0 release

**Pull request merged**